src/app_conn_pool.rs

```diff
@@ -197,13 +197,13 @@ impl AppConnPool {
         tracing::info!("Connecting to app websocket at {}:{}", host, app_port);
 
         // Build a connection request
-        let request =
-            ConnectRequest::from(format!("{host}:{app_port}").parse::<SocketAddr>().map_err(
-                |e| {
-                    HcHttpGatewayError::ConfigurationError(format!("Invalid socket address: {}", e))
-                },
-            )?)
-            .try_set_header("Origin", HTTP_GW_ORIGIN)?;
+        let request = ConnectRequest::from(
+            format!("{host}:{app_port}")
+                .parse::<SocketAddr>()
+                .expect("Host and app port must be valid"),
+        )
+        .try_set_header("Origin", HTTP_GW_ORIGIN)
+        .expect("Origin headers have gone out of fashion");
```

Comment:
This looks pretty bold, but we're already expecting the host address earlier and the port is also assured. Regarding headers, "Origin" is always safe to set. Or do you prefer to re-introduce std::io::Errors in the error enum and return a 500 error to callers (in theory, because it'll never happen in practice)?

Reply:
This is a step in the right direction. I think we don't need to be messing with strings at all at this point at all. We should resolve to an IP address as soon as we get the input configuration and then just be passing a Holochain IP around instead of strings. I've commented about this on a couple of PRs now but don't want to change too much at once :D